src/Ast.ml

```diff
@@ -1675,7 +1675,8 @@ end = struct
             match ctx_desc with
             | Pexp_apply
                 ({pexp_desc= Pexp_ident {txt= Lident i}}, _ :: (_, e2) :: _)
-              when e2 == exp && is_infix_id i ->
+              when e2 == exp && is_infix_id i
+                   && Poly.(prec_ast ctx < Some Apply) ->
```

Comment:
Thinking about this again, I'm not sure my suggestion isn't risky. It would be safer to check that `prec_ast ctx` is `Some p` where `p < Apply`.